file/delete_scheduler.h

```diff
@@ -45,6 +45,7 @@ class DeleteScheduler {
   // Set delete rate limit in bytes per second
   void SetRateBytesPerSecond(int64_t bytes_per_sec) {
     rate_bytes_per_sec_.store(bytes_per_sec);
+    MaybeCreateBackgroundThread();
```

Comment:
It would be safer to lock the mutex before calling this. I don't expect SetRateBytesPerSecond() to be called by multiple threads, but it doesn't hurt to handle it anyway.

Reply:
You are right. Rocksdb maybe crash If we call the SetRateBytesPerSecond() function concurrently without lock. Unfortunately, this code seems to have been merged without fix on it.